### PR TITLE
Add authentication

### PR DESCRIPTION
**Pull request by alloy**
emission-authentication

### CODE REVIEW

**Comment by alloy** [comment]
@sarahscott @mennenia @orta You can delete the `Example/Emission/Configuration.h` file.

File: Example/Emission/Configuration.h.sample [hunk 1]
```diff
@@ -1,19 +0,0 @@
...
-//   $ curl -s -H 'x-access-token: [OAUTH_TOKEN]' "https://stagingapi.artsy.net/api/v1/me" | ruby -r json -e 'puts JSON.parse(ARGF.read)["_id"]' | tee /dev/tty | pbcopy
-//
-// Again, your user ID will be placed on your pasteboard, so you can directly paste it below in the `USER_ID` macro.
-//
-//
-// (The client ID & secret are of Eigen OSS: https://github.com/artsy/eigen/blob/0e193d1b/Makefile#L36-L37)
-
-#define USER_ID @"USER ID GOES HERE"
-
-#define OAUTH_TOKEN @"TOKEN GOES HERE"
```
Comment (orta) [inline Example/Emission/Configuration.h.sample]: aww, this was such a nice file

Comment (alloy) [inline Example/Emission/Configuration.h.sample]: Agreed. To all good things must come an end, so I had to put an end to it.

File: Example/Podfile [hunk 2]
```diff
@@ -7,19 +7,23 @@ target 'Emission' do
...
+  pod 'SAMKeychain'
+  pod 'Artsy+Authentication', :subspecs => %w(email),
+                              :git => 'https://github.com/artsy/Artsy-Authentication.git',
+                              :branch => 'fetch-user-details'
 
   target 'EmissionTests' do
     inherit! :search_paths
 
-    pod 'React', :path => react_path, :subspecs => ['RCTTest']
+    pod 'React', :path => react_path, :subspecs => %w(RCTTest)
```
Comment (orta) [inline Example/Podfile]: I never really got the preference for `%w` over the canonical `[` & `]`

**Comment by orta** [comment]
👍

File: Example/Podfile (hunk 2 above)
Comment (alloy) [inline Example/Podfile]: Less characters to type 😄

Comment (alloy) [inline Example/Podfile]: (Not in the case of a single word, ofc, but this is just for consistence with the other line.)

File: Example/Emission/Configuration.h.sample (hunk 1 above)
Comment (mennenia) [inline Example/Emission/Configuration.h.sample]: t'was a good'n indeed

**Pull request merged**